src/StudentOrganizer.Infrastructure/Services/TeamService.cs

```diff
@@ -71,5 +75,49 @@ public async Task UpdateTeamName(UpdateTeamName command)
 
 			await _groupRepository.SaveChangesAsync();
 		}
-	}
-}
\ No newline at end of file
+
+        public async Task AddAssignment(AddAssignment command)
+        {
+            await _administratorService.ValidateAtLeastModerator(command.UserId, command.GroupId);
+
+            var assignment = await _assignmentRepository.GetAsync(command.AssignmentId);
+
+            if(assignment == null)
+            {
+                throw new AppException($"Assignment {command.AssignmentId} doesn't exist", AppErrorCode.DOESNT_EXIST);
+            }
+
+            var team = await _teamRepository.GetAsync(command.TeamName);
+
+            if(team == null)
+            {
+                throw new AppException($"Team {command.TeamName} doesn't exist", AppErrorCode.DOESNT_EXIST);
+            }
+
+            team.AddAssignment(assignment);
+            await _teamRepository.UpdateAsync(team);
+        }
+
+        public async Task DeleteAssignment(DeleteAssignment command)
+        {
+            await _administratorService.ValidateAtLeastModerator(command.UserId, command.GroupId);
+
+            var assignment = await _assignmentRepository.GetAsync(command.AssignmentId);
+
+            if (assignment == null)
+            {
+                throw new AppException($"Assignment {command.AssignmentId} doesn't exist", AppErrorCode.DOESNT_EXIST);
+            }
+
+            var team = await _teamRepository.GetAsync(command.TeamName);
```

Comment:
To nie przejdzie nazwy teamów nie są unikatowe same w sobie, są unikatowe tylko w grupie. Napisz taką metodę w repo która pobierze Grupę z teamami i assignmentami i na tym już działaj